jetty-util/src/main/java/org/eclipse/jetty/util/Jetty.java

```diff
@@ -18,14 +18,51 @@
 
 package org.eclipse.jetty.util;
 
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+import java.io.InputStream;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Properties;
+
 public class Jetty
 {
+    private static final Logger LOG = Log.getLogger( Jetty.class);
+
     public static final String VERSION;
     public static final String POWERED_BY;
     public static final boolean STABLE;
+    public static final String GIT_HASH;
+
+    /**
+     * a formatted build timestamp with pattern yyyy-MM-dd'T'HH:mm:ssXXX
+     */
+    public static final String BUILD_TIMESTAMP;
+    private static Properties BUILD_PROPERTIES = new Properties( );
 
     static
     {
+        try
+        {
+            try (InputStream inputStream = //
+                     Jetty.class.getResourceAsStream( "/org/eclipse/jetty/version/build.properties" ))
+            {
+                BUILD_PROPERTIES.load( inputStream );
+            }
+        }
+        catch ( Exception e )
+        {
+            // ignore this
+            LOG.ignore( e );
+        }
+
+        GIT_HASH = BUILD_PROPERTIES.getProperty( "buildNumber", "unknown" );
+        System.setProperty( "jetty.git.hash" , GIT_HASH );
+        BUILD_TIMESTAMP = formatTimestamp(BUILD_PROPERTIES.getProperty( "timestamp", "unknown" ));
+
+        // olamy using BUILD_PROPERTIES.getProperty("version") will contain version from the pom
```

Comment:
I don't understand this comment? Also no need for user name in comments, that is what git is for

Reply:
my concern is the code here whereas it can be replace by a single line now. But I don't have the story of this :-)